### PR TITLE
Hide facility column results table 

### PR DESCRIPTION
**Pull request by johanna-skylight**
# FRONTEND PULL REQUEST

## Related Issue

Fixes [3691](https://github.com/CDCgov/prime-simplereport/issues/3691)

## Changes Proposed

- Results table logic now accepts the flag displayFacility. When this flag is true the Facility Column will be visible and when is false the column will be hidden.
- The column header got renamed to Facility
- The flag displayFacilityColumn was added to the main view/component.

## Additional Information
No new test coverage was required.
I also moved the graphql queries out of the testResultList.tsx file and into their own file as the tsx file is getting too long. It should be considered if we want to adopt this (creating a specific file for a container component's queries) as a standard for the app.

## Testing
Load an account with multiple facilities. 
Select all facilities in the filter area and see that the column "Facility" displays in the table.
Select a facility and see that the column "Facility" column does not display.
 
## Screenshots / Demos
**all facilities**
Screen Shot 2022-05-25 at 3 26 49 PM

**testing facility**
Screen Shot 2022-05-25 at 3 26 57 PM

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### CODE REVIEW

**Comment by nathancrtr** [comment]
> I also moved the graphql queries out of the testResultList.tsx file and into their own file as the tsx file is getting too long. It should be considered if we want to adopt this (creating a specific file for a container component's queries) as a standard for the app.

Very good thought! In fact you've arrived at the exact solution we've been moving towards. GraphQL queries should live in an `operations.graphql` in the relevant subdirectory (in this case, `frontend/src/app/testResults`).

**Comment by johanna-skylight** [comment]
@nathancrtr I just added instructions on how to handle graphql query implementations to the repo's wiki. I think it is important enough to document. Let me know if the description looks good! https://github.com/CDCgov/prime-simplereport/wiki/SimpleReport-Style-Guide

**Comment by johanna-skylight** [comment]
@kenieh I just added screenshots with the multiplex columns

**Comment by nathancrtr** [comment]
> @nathancrtr I just added instructions on how to handle graphql query implementations to the repo's wiki. I think it is important enough to document. Let me know if the description looks good! https://github.com/CDCgov/prime-simplereport/wiki/SimpleReport-Style-Guide

This is great, thanks! The wiki is criminally underutilized but it's a perfect place to document these kinds of processes.

File: frontend/src/app/testResults/TestResultsList.tsx [hunk 1]
```diff
@@ -820,11 +822,14 @@ const TestResultsList = () => {
         : filterFacilityId || activeFacilityId,
     ...queryParams,
   };
-
-  const count = useQuery(resultsCountQuery, {
+  const count = useGetResultsCountByFacilityQuery({
     fetchPolicy: "no-cache",
     variables: countQueryVariables,
   });
+  /*const count = useQuery(RESULTS_COUNT_QUERY, {
```
Comment (emmastephenson) [inline frontend/src/app/testResults/TestResultsList.tsx:829]: This comment can probably be removed

**Review by emmastephenson — APPROVED** [review]
Overall LGTM! Thanks for adding to the wiki as well.

File: frontend/src/app/testResults/TestResultsList.tsx (hunk 1 above)
Comment (johanna-skylight) [reply frontend/src/app/testResults/TestResultsList.tsx]: Thanks for catching this 😮 !!! I removed it

**Review by emmastephenson — DISMISSED** [review]

File: frontend/src/app/testResults/TestResultsList.test.tsx [hunk 2]
```diff
@@ -1488,7 +1488,7 @@ describe("TestResultsList", () => {
     expect(await row.findByText("Colleer, Barde X")).toBeInTheDocument();
     expect(await row.findByText("DOB: 11/07/1960")).toBeInTheDocument();
     expect(await row.findByText("Negative")).toBeInTheDocument();
-    expect(await row.findByText("Facility 1")).toBeInTheDocument();
+    expect(row.queryByText("Facility 1")).not.toBeInTheDocument();
```
Comment (fzhao99) [inline frontend/src/app/testResults/TestResultsList.test.tsx:1491]: I would prefer we add a test here for the happy path case on multiple facilities (ie asserting against the facility column/respective names showing up when we're passing 2+ facilities to the front end.) Don't want an inadvertent code change in the future to cause the facility column to not show up or the various facility names to appear incorrectly. Happy to explain more if needed!

File: frontend/src/app/testResults/TestResultsList.tsx [hunk 3]
```diff
@@ -90,15 +95,17 @@ const tableHeaders = (hasMultiplexResults: boolean) => (
...
-    <th scope="col" className="test-facility-cell">
-      Testing facility
-    </th>
     <th scope="col" className="test-device-cell">
       Test device
     </th>
     <th scope="col" className="submitted-by-cell">
       Submitted by
     </th>
+    {displayFacility && (
```
Comment (fzhao99) [inline frontend/src/app/testResults/TestResultsList.tsx:104]: nit: for the sake of consistency/style, we might consider using the same tenary syntax used on line 88 in the event of displayFacility being false. Similar feedback for line 236.

File: frontend/src/generated/graphql.tsx [hunk 4]
```diff
@@ -2562,6 +2548,20 @@ export type GetAllFacilitiesQuery = {
     | undefined;
 };
 
+export type GetResultsCountByFacilityQueryVariables = Exact<{
```
Comment (fzhao99) [inline frontend/src/generated/graphql.tsx:2551]: Thanks for moving this out into its own file/adding things to the wiki!

**Review by fzhao99 — CHANGES_REQUESTED** [review]
left a comment about adding a test case for the happy path of 2+ facilities, since right now, we're only asserting against something not being in the document in the tests for TestResultsList. Let me know if you want to discuss further!

**Review by kenieh — APPROVED** [review]
LGTM! thanks for working on this

**Pull request merged**